L1Trigger/L1TCommon/interface/Parameter.h

```diff
@@ -0,0 +1,160 @@
+#ifndef L1Trigger_L1TCommon_l1t_Parameter_h
+#define L1Trigger_L1TCommon_l1t_Parameter_h
+
+#include<map>
+#include<list>
+#include<vector>
+#include<string>
+#include<iostream>
+#include<algorithm>
+#include<stdexcept>
+#include<type_traits>
+#include<cstring>
+
+namespace l1t {
+
+// assuming string is castable to T (some specific cases are defined in the end)
+template<class T> T castTo(const char *arg);
+
+class Parameter {
+private:
+    std::string id, type, procOrRole;  // attributes
+    std::string scalarOrVector, delim; // setting could be one single parameter
+    std::map<std::string, std::vector<std::string> > table; // or a map of columns, no multimap here: column names are assumed be unique!
+    std::map<std::string,unsigned int> columnNameToIndex; // remember original positions of the columns
+
+public:
+    std::string getId        (void) const noexcept { return id;             }
+    std::string getProcOrRole(void) const noexcept { return procOrRole;     }
+    std::string getType      (void) const noexcept { return type;           }
+    std::string getValueAsStr(void) const noexcept { return scalarOrVector; }
+
+    bool isScalar(void) const noexcept {
+        if( type.find("vector") != std::string::npos ||
```

Comment:
@davidlange6 I am not sure I understand what you mean here. Non of the l1t::Parameter objects are constructed or manipulated at compile-time. They are constructed as the xml configuration parsed in runtime. So declaring any of the trivial methods of this class as constexpr won't have any effect in the end.